testSuite/scripts/test_upload_block_blob.py

```diff
@@ -504,15 +504,15 @@ def test_download_blob_include_flag(self):
             x = json.loads(result, object_hook=lambda d: namedtuple('X', d.keys())(*d.values()))
         except:
             self.fail('error parsing the output in Json Format')
-        self.assertEquals(x.TransfersCompleted, 3)
+        self.assertEquals(x.TransfersCompleted, 6)
```

Comment:
It seems weird that we didn't change the test at all, and the test only includes 3 files with the include pattern, but now we are downloading 6 files instead of 3. What's up there?

Reply:
So, include and exclude previously only did things on the root directory (because they were checking the relative path, not the file name).

We ended up declaring this a bug and that the include and exclude filters should affect file names under all sub directories.

Reply:
As a result, there are 3 matching files under each directory. Therefore, 6 files get included.